cpp/velox/utils/ConfigExtractor.cc

```diff
@@ -41,4 +57,103 @@ bool debugModeEnabled(const std::unordered_map<std::string, std::string>& confMa
       std::make_shared<const facebook::velox::core::MemConfigMutable>(confMap);
   return veloxCfg->get<bool>(kDebugModeEnabled, false);
 }
+
+std::shared_ptr<facebook::velox::core::MemConfigMutable> getHiveConfig(
+    const std::shared_ptr<const facebook::velox::Config>& conf) {
+  auto hiveConf = std::make_shared<facebook::velox::core::MemConfigMutable>();
+
+#ifdef ENABLE_S3
+  std::string awsAccessKey = conf->get<std::string>("spark.hadoop.fs.s3a.access.key", "");
+  std::string awsSecretKey = conf->get<std::string>("spark.hadoop.fs.s3a.secret.key", "");
+  std::string awsEndpoint = conf->get<std::string>("spark.hadoop.fs.s3a.endpoint", "");
+  bool sslEnabled = conf->get<bool>("spark.hadoop.fs.s3a.connection.ssl.enabled", false);
+  bool pathStyleAccess = conf->get<bool>("spark.hadoop.fs.s3a.path.style.access", false);
```

Comment:
We need to put these configurations in a resource file and shared by native and scala, then generate a header file and scala file. We may do this in another PR.